### PR TITLE
introduce typescript

### PR DESCRIPTION
**Pull request by daiki44**
# What?
- Introduce TypeScript
- Refactoring (remove some unused files)

# TODO
- Add types to each file

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: npm i --frozen-lockfile
+      - name: lint
+        run: |
+          npm run lint
```
Comment (tkmct) [inline .github/workflows/lint.yml:22]: add tsc check

File: src/client.ts [hunk 2]
```diff
@@ -1,39 +1,54 @@
+import LightClient from '@cryptoeconomicslab/plasma-light-client'
 import initialize from './initialize'
+import { Web3Wallet, WALLET_KIND } from './wallet'
 
 class ClientWrapper {
+  private instance: LightClient | null
```
Comment (tkmct) [inline src/client.ts:6]: what about just `private instance?: LightClient`?

File: src/client.ts [hunk 3]
```diff
@@ -1,39 +1,54 @@
...
+   * @returns {?Client}
+   */
+  get client(): LightClient | null {
+    return this.instance
+  }
+
+  /**
+   * Returns Web3Wallet.
+   */
+  get wallet(): Web3Wallet | null {
```
Comment (tkmct) [inline src/client.ts:49]: you can just make wallet readonly.

File: src/client.ts [hunk 4]
```diff
@@ -1,39 +1,54 @@
...
     if (!this.instance) return
     await this.instance.start()
   }
+
+  /**
+   * Returns client singleton. Lazily initialized on client side.
+   * Returns null on server side.
+   * @returns {?Client}
+   */
+  get client(): LightClient | null {
```
Comment (tkmct) [inline src/client.ts:42]: you can just make client readonly.

File: src/store/transfer.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+const initialState: State = {
+  transferredToken: config.PlasmaETH,
+  transferredAmount: '',
+  recepientAddress: '',
+  transferPage: 'confirmation-page',
+  status: TRANSFER_PROGRESS.INITIAL,
+  error: null
+}
+
+interface TransferAction {
```
Comment (tkmct) [inline src/store/transfer.ts:47]: Let's create generic Action Type. Use specific ActionTypeStrings.

File: src/client.ts (hunk 4 above)
Comment (daiki44) [reply src/client.ts]: I can't set the value in the constructor.
That's why it is a private field.
https://github.com/cryptoeconomicslab/gazelle-wallet/blob/feat/wak-398/ts/src/client.ts#L14

Do you know how to set them as readonly fields?

Comment (tkmct) [reply src/client.ts]: Ah, okay.
Then maybe it's better to use factory pattern.

**Comment by daiki44** [comment]
updated! 
38c2b68
fc0ea04

**Review by tkmct — APPROVED** [review]
LGTM

**Pull request merged**